mne/preprocessing/_peak_finder.py

```diff
@@ -50,11 +50,26 @@ def peak_finder(x0, thresh=None, extrema=1, verbose=None):
     s = x0.size
 
     if x0.ndim >= 2 or s == 0:
-        raise ValueError('The input data must be a non empty 1D vector')
+        raise ValueError('The input data must be a non-empty 1D vector')
+
+    if x0.dtype.kind in 'uib':  # integer dtype
+        x0 = x0.astype(float)
+
+    eps = 1e2 * np.finfo(x0.dtype).eps
+
+    if thresh is not None:
+        assert thresh >= 0.
 
     if thresh is None:
         thresh = (np.max(x0) - np.min(x0)) / 4
+        if thresh < eps:
+            thresh = eps
         logger.debug('Peak finder automatic threshold: %0.2g' % (thresh,))
+    elif thresh < eps:
+        raise ValueError(
+            f'The selected threshold value is too small for peak detection '
+            f'due to inevitable floating-point imprecisions.\nPlease pick a '
+            f'value >= {eps:g} (You passed thresh={thresh})')
```

Comment:
I am not convinced this is right. We have to keep in mind `eps` is relative to 1., it's not an absolute minimum delta between floats (that would be `finfo(x0.dtype).tiny`, and even that's before going denormalized). For example consider this on `main`, which seems correct:
```
>>> import mne
>>> import numpy as np 
>>> x = np.full(1000, 1e-20)
>>> x[500] = 2e-20
>>> mne.preprocessing.peak_finder(x)
(array([500]), array([2.e-20]))
```
But we have
```
>>> (np.max(x) - np.min(x)) / 4
2.5e-21
>>> 1e2 * np.finfo(x.dtype).eps
2.220446049250313e-14
```
So this PR will find nothing significant.

For EEG/EOG data in V, this might be okay. For data in T or T/M, these hang out around 1e-12 so this can become problematic.